datamap/bloom/src/main/java/org/apache/carbondata/datamap/bloom/BloomDataMapWriter.java

```diff
@@ -69,6 +86,27 @@
     indexBloomFilters = new ArrayList<>(indexColumns.size());
     initDataMapFile();
     resetBloomFilters();
+
+    keyGenerator = segmentProperties.getDimensionKeyGenerator();
```

Comment:
Can we optimize this instead of passing the whole `SegmentProperties` into this Writer class? Please check @ravipesala

Reply:
Here we use the keyGenerator, ColumnarSpitter and dimensions for this segment.